packages/protocol/contracts/governance/Election.sol

```diff
@@ -851,22 +851,30 @@ contract Election is
     // Holds the number of members elected for each of the eligible validator groups.
     uint256[] memory numMembersElected = new uint256[](electionGroups.length);
     uint256 totalNumMembersElected = 0;
-    // Assign a number of seats to each validator group.
-    while (totalNumMembersElected < maxElectableValidators) {
-      uint256 groupIndex = 0;
-      bool memberElected = false;
-      (groupIndex, memberElected) = dHondt(
-        electionGroups,
-        numMembers,
-        totalNumMembersElected,
-        numMembersElected
-      );
 
-      if (memberElected) {
+    uint256[] memory keys = new uint256[](electionGroups.length);
+    for (uint256 i = 0; i < electionGroups.length; i++) {
+      keys[i] = i;
+    }
+    FixidityLib.Fraction[] memory groupVotes = new FixidityLib.Fraction[](electionGroups.length);
+    for (uint256 i = 0; i < electionGroups.length; i++) {
+      groupVotes[i] = FixidityLib.newFixed(votes.total.eligible.getValue(electionGroups[i]));
+    }
+
+    // Assign a number of seats to each validator group.
+    while (totalNumMembersElected < electableValidators.max && electionGroups.length > 0) {
+      uint256 groupIndex = keys[0];
+      if (groupVotes[groupIndex].unwrap() == 0) break;
+      if (numMembers[groupIndex] <= numMembersElected[groupIndex]) {
```

Comment:
It would be helpful to have comments here that explain the intuition like previously